### ISSUE TITLE
Candidates not displaying

### ISSUE THREAD

**Issue by wpcarro** [aaronjensen/company-flow#11]
I'm having trouble getting the autocompletion candidates to display. I've tried to manually invoke `company-flow`, but I get the following error message: `Cannot complete at point`.

I tried reading through the `company-flow.el` source, and I called `(company-flow--candidates-query nil (lambda (x) (print x)))` in the `Eval: ` buffer. This produced a list of candidates that matches the list I get when I run:

```
$ flow autocomplete LINE COL <FILE
```

I'm having a tough time debugging this, however. Any help would be greatly appreciated.

**Comment by aaronjensen** [comment]
Hi @wpcarro what does `M-x company-diag` show when run after putting the point after `foo.` here:

```js
// @flow
const foo: Array<number> = [3]
foo.
```

**Comment by wpcarro** [comment]
```
Emacs 25.1.1 (x86_64-apple-darwin13.4.0) of 2016-09-20 on builder10-9.porkrind.org
Company 0.9.4

company-backends: (company-bbdb company-nxml company-css company-eclim company-semantic company-clang company-xcode company-cmake company-capf company-files
              (company-dabbrev-code company-gtags company-etags company-keywords)
              company-oddmuse company-dabbrev)

Used backend: (company-dabbrev-code company-gtags company-etags company-keywords)

Major mode: rjsx-mode
Prefix: ""
Completions:
  "Section"
  "default"
  "export"
  "section"
  "Item"
  "_item"
  "items"
  "title"
  "section"
  "Props"
  "items"
  "title"
  "Section"
  "const"
  "Array"
  "items"
  "string"
  "title"
  "Props"
  "type"
  "from"
  "React"
  "import"
  "number"
  "Array"
  "const"
```

**Comment by aaronjensen** [comment]
It looks like you're not using the backend. Since you're using rjsx-mode, you need to do:

```
(add-to-list 'company-flow-modes 'rjsx-mode)
```

I should add that as a default supported mode.

**Comment by aaronjensen** [comment]
Here you go: https://github.com/aaronjensen/company-flow/commit/b226369710b0f1fa79f5fd6d70086158400f3427

Let me know if adding the mode helps. Also, be sure you follow the instructions in the readme and:

```
(eval-after-load 'company
  '(add-to-list 'company-backends 'company-flow))
```

I didn't see `company-flow` in your list above.

**Comment by wpcarro** [comment]
I'm now appending `rjsx-mode` to `company-flow-modes`, but I'm not seeing any changes. Here's my `init.el`

```
(use-package company-flow
  :config
  (add-to-list 'company-flow-modes 'rjsx-mode)
  (with-eval-after-load 'company
    '(add-to-list 'company-backends 'company-flow)))
```

Here's the updated output from running `company-diag` after `foo.`:

```
Emacs 25.1.1 (x86_64-apple-darwin13.4.0) of 2016-09-20 on builder10-9.porkrind.org
Company 0.9.4

company-backends: (company-bbdb company-nxml company-css company-eclim company-semantic company-clang company-xcode company-cmake company-capf company-files
              (company-dabbrev-code company-gtags company-etags company-keywords)
              company-oddmuse company-dabbrev)

Used backend: (company-dabbrev-code company-gtags company-etags company-keywords)

Major mode: rjsx-mode
Prefix: ""
Completions:
  "number"
  "Array"
  "const"
```

**Comment by wpcarro** [comment]
I'm running the following in a `*scratch*` to verify that the append operation worked.

```elisp
company-flow-modes
;; => (rjsx-mode js-mode js-jsx-mode js2-mode js2-jsx-mode web-mode)
```

**Comment by aaronjensen** [comment]
I don't know how the rest of your emacs is set up, so I don't know if you have something else that is overriding `company-backends`. 

Your `company-backends` does not include `company-flow`, so it will not work. In your buffer, try to eval (`M-:`) `(add-to-list 'company-backends 'company-flow)` and then rerun company-diag.

It should be in the list of both `company-backends` and `Used backend:`

**Comment by wpcarro** [comment]
From your README, I added: 


```elisp
(with-eval-after-load 'company
    '(add-to-list 'company-backends 'company-flow))
```

Turns out the leading quote before `add-to-list` is problematic and preventing `company-flow` from being registered in `company-backends`. Not sure if that's an error on your end or if your README is incompatible with my setup. Worth pointing out though!

After removing the leading quote and doing:

```elisp
(with-eval-after-load 'company
    (add-to-list 'company-backends 'company-flow))
```

`company-flow` is including in `company-diag` and things seem to be working!

Post Mortem:
1. add `rjsx-mode` to `company-flow-modes`
2. remove leading quote from `add-to-list` call
